frontend/src/app/store/simple-store.ts

```diff
@@ -0,0 +1,34 @@
+import { BehaviorSubject } from 'rxjs';
+import { Store } from './types';
+import { Params } from '@angular/router';
+import * as _ from 'lodash';
+import { map, scan } from 'rxjs/operators';
+import { omitNullParameters } from '../utils/params';
+
+/** simple store that does not depend on services or routing
+ *
+ * Stores parameters in an internal BehaviorSubject. Can be used
+ * for testing or to manage stand-alone models.
+ */
+export class SimpleStore implements Store {
+    paramUpdates$ = new BehaviorSubject<Params>({});
+    params$ = new BehaviorSubject<Params>({});
+
+    constructor() {
+        this.paramUpdates$.pipe(
+            scan(this.merge),
+            map(omitNullParameters),
+            map(obj => _.mapValues(obj, _.toString))
+        ).subscribe(params =>
+            this.params$.next(params)
+        );
+    }
```

Comment:
Alright, here I understand the distinction between `paramUpdate$` and `param$`. Nice use of the `scan` function. Elegant! I'm not sure I agree with emulating the `router-store`'s behaviour here and converting everything to `string`. Should the router store not implement functions to infer & convert to `number` or `boolean` types instead?

Reply:
That sounds nice, but I don't think that can be done in a way where valid strings like `'5'` or `'true'` are preserved _and_ the routes remain backwards compatible.

Reply:
I thought we'd always be able to convert back and fro, but of course user queries may include strings like '5' or 'true'. Hmm...

Reply:
The current solution that this conversion is left to the model implementation, which knows what type to expect. For instance, the `BooleanFilter` knows that `true` should be parsed as a boolean value, but for the query parameter, this would be interpreted as the string `'true'`.